File: python/38.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        prev = self.countAndSay(n-1)
        count = 1
        curr = ""
        for i in range(len(prev)):
            if i == len(prev)-1 or prev[i] != prev[i+1]:
                curr += str(count) + prev[i]
                count = 1
            else:
                count += 1
        return curr

File: C#/38.cs
public class Solution {
    public string CountAndSay(int n) {
        if(n == 1) return "1";
        string prev = CountAndSay(n-1);
        string result = "";
        int count = 1;
        for(int i=0; i<prev.Length; i++){
            if(i == prev.Length-1 || prev[i] != prev[i+1]){
                result += count.ToString() + prev[i];
                count = 1;
            }
            else{
                count++;
            }
        }
        return result;
    }
}